### PR TITLE
Papp 736 add query ordering support to specification pattern

### PR DESCRIPTION
**Pull request by Maxim-Zavidei**
# Overview:
* [PAPP-736] Added ordering functionality to the specification pattern
* [PAPP-736] Implemented a method to retrieve lastest sensor update
* [PAPP-736] Inlcude the sensor installation with the bundle

[PAPP-736]: https://papp.atlassian.net/browse/PAPP-736?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PAPP-736]: https://papp.atlassian.net/browse/PAPP-736?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PAPP-736]: https://papp.atlassian.net/browse/PAPP-736?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Papp.Persistence/Source/DataAccess/SensorUpdate/ISensorUpdateDataAccess.cs [hunk 1]
```diff
@@ -7,4 +7,9 @@ namespace Papp.Persistence.DataAccess;
 /// </summary>
 public interface ISensorUpdateDataAccess : IGenericDataAccess<SensorUpdate>
 {
+    /// <summary>
+    /// Retrieves last Sensor Update of a specific sensor id.
+    /// </summary>
+    /// <returns>Latest Sensor Update or null if sensor had no registered updates.</returns>
+    SensorUpdate GetLastestBySensorId(string id);
```
Comment (Logstor) [inline Papp.Persistence/Source/DataAccess/SensorUpdate/ISensorUpdateDataAccess.cs:14]: Shouldn't be a nullable type or is this turned off for this project? :slightly_smiling_face:

File: Papp.Persistence/Source/DataAccess/SensorUpdate/SensorUpdateDataAccess.cs [hunk 2]
```diff
@@ -11,4 +11,12 @@ public SensorUpdateDataAccess(PappDbContext context) : base(context)
     {
         this.context = context;
     }
+
+    public SensorUpdate? GetLastestBySensorId(string id)
```
Comment (Logstor) [inline Papp.Persistence/Source/DataAccess/SensorUpdate/SensorUpdateDataAccess.cs:15]: Remember `///<inheritdoc/>` :+1:

**Review by Logstor — CHANGES_REQUESTED** [review]
Really nice :+1: 
Only a little documentation thingy :wink:

File: Papp.Persistence/Source/DataAccess/SensorUpdate/ISensorUpdateDataAccess.cs (hunk 1 above)
Comment (Maxim-Zavidei) [reply Papp.Persistence/Source/DataAccess/SensorUpdate/ISensorUpdateDataAccess.cs:14]: oh yes that's correct

File: Papp.Persistence/Source/DataAccess/SensorUpdate/SensorUpdateDataAccess.cs (hunk 2 above)
Comment (Maxim-Zavidei) [reply Papp.Persistence/Source/DataAccess/SensorUpdate/SensorUpdateDataAccess.cs:15]: good notice

File: Papp.Persistence/Source/DataAccess/SensorUpdate/SensorUpdateDataAccess.cs [hunk 3]
```diff
@@ -11,4 +11,13 @@ public SensorUpdateDataAccess(PappDbContext context) : base(context)
...
         this.context = context;
     }
+
+    /// <inheritdoc/>
+    public SensorUpdate? GetLastestBySensorId(string id)
+    {
+        return this.context.SensorUpdates
+            .Where(e => e.SensorId.Equals(id))
+            .OrderByDescending(e => e.Ts)
+            .FirstOrDefault();
```
Comment (Logstor) [inline Papp.Persistence/Source/DataAccess/SensorUpdate/SensorUpdateDataAccess.cs:21]: Maybe also use the `take(1)` here :wink:

**Review by Logstor — CHANGES_REQUESTED** [review]
One small thing, otherwise really good work :muscle:

**Pull request merged**